From: Johannes Thumshirn <jthumshirn@suse.de>
Date: Mon  8 Jul 11:05:25 CEST 2019
Subject: Kabi fixup blk_mq_register_dev()
References: bsc#1140637
Patch-mainline: Never, KABI consistency

This is a kabi consistency patch for
patches.fixes/blk-mq-free-hw-queue-s-resource-in-hctx-s-release-handler.patch.

patches.fixes/blk-mq-free-hw-queue-s-resource-in-hctx-s-release-handler.patch
adds "block/blk.h" to block/blk-mq-sysfs.c in order to get the definitions for
blk_free_flush_queue().

But adding "block/blk.h" breaks the kABI for blk_mq_register_dev() by making
it more defined.

Fix this by removing the include and pulling the definition in as an "extern"

Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 block/blk-mq-sysfs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/block/blk-mq-sysfs.c
+++ b/block/blk-mq-sysfs.c
@@ -10,7 +10,6 @@
 #include <linux/smp.h>
 
 #include <linux/blk-mq.h>
-#include "blk.h"
 #include "blk-mq.h"
 #include "blk-mq-tag.h"
 
@@ -20,6 +19,7 @@ static void blk_mq_sysfs_release(struct
 
 static void blk_mq_hw_sysfs_release(struct kobject *kobj)
 {
+	extern void blk_free_flush_queue(struct blk_flush_queue *q);
 	struct blk_mq_hw_ctx *hctx = container_of(kobj, struct blk_mq_hw_ctx,
 						  kobj);
 
